handler/actions.go

```diff
@@ -568,27 +624,89 @@ func (h *Handler) prune(ea *EventAction) error {
 		return err
 	}
 
+	matches := h.getMatches(ea.Action, ev.Text)
+
+	var nmenv []string
+	for _, m := range matches {
+		if strings.Contains(m, "|") {
+			nmenv = strings.Split(m, "|")
+		} else {
+			// No env
+			nmenv = append(nmenv, "")
+			nmenv = append(nmenv, m)
+		}
+	}
+
 	resources := h.data.GetResources()
+	removedResource := false
 	for _, res := range resources {
-		q, err := h.data.GetQueueForResource(res.Name, res.Env)
-		if err != nil {
-			// this shouldn't happen, but there's nothing to alert the user to
-			log.Errorf("%+v", err)
-			continue
-		}
+		if (nmenv[0] == res.Env) && (nmenv[1] == res.Name) {
```

Comment:
nit: you could save an indentation level here with:
```suggestion
		if (nmenv[0] != res.Env) || (nmenv[1] != res.Name) {
		    continue
		}
```

Reply:
done!